public/javascripts/PSMap/AddNeighborhoodsToMap.js

```diff
@@ -168,14 +170,22 @@ function AddNeighborhoodsToMap(map, neighborhoodGeoJSON, completionRates, labelC
                 const regionCenter = turf.centerOfMass(currRegion).geometry.coordinates;
                 neighborhoodTooltip.setLngLat({ lng: regionCenter[0], lat: regionCenter[1] }).addTo(map);
 
-                // Add listeners to popup so the popup closes when the mouse leaves the popup area.
-                neighborhoodTooltip._content.onmouseout = function (e) {
-                    if (!e.toElement || !e.toElement.closest('.mapboxgl-popup')) {
+                // Clear timeout when entering new toolTip
+                neighborhoodTooltip._content.onmouseenter = function () {
+                    if (tooltipTimeout) {
+                        clearTimeout(tooltipTimeout); // Clear the timeout if the mouse enters the tooltip
```

Comment:
And when we call `clearTimeout` we're actually never unsetting the `tooltipTimeout` variable. So after the first tooltip opens, `tooltipTimeout` will never be falsey. Which is fine, because we can keep trying to clear the timeout for a timer that's already been cleared. Similarly, we can keep calling `clearTimeout` on `undefined`! So maybe let's just not bother with checking if `tooltipTimeout` is defined to simplify things even more:
```
                // Clear timeout when entering a tooltip.
                neighborhoodTooltip._content.onmouseenter = function () { clearTimeout(tooltipTimeout); };
```

And we can simplify in the same way throughout

Reply:
Same sort of situation happens later in the code that still needs to be redone:
```
        map.on('mouseenter', NEIGHBORHOOD_LAYER_NAME, () => {
            if (tooltipTimeout) {
                clearTimeout(tooltipTimeout);
            }
```